### PR TITLE
Reduce libguestfs image size

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Exclude quite a bunch of RPMs which are not needed by libguestfs but are pulled in automatically due to RPM dependencies. Reduces the size of pulled in dependencies from roughly 1.3 GB to 500 M.

More dependencies can be removed. This is just a first pass. Thanks @alicefr for providing initial investigations on potential candidates for removal.

This step is needed to achieve two goals:
 * keep the production image small for users
 * speed up kubevirt-build, since this image drastically slows down builds and deploys. Especially when nothing is cached.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
@andreabolognani FYI: We just do this step now already before #5991  because the image makes building significantly slower.

File: hack/rpm-deps.sh [hunk 1]
```diff
@@ -102,7 +102,12 @@ bazel run \
     //:bazeldnf -- rpmtree --public --name libguestfs-tools \
     $basesystem \
     libguestfs \
-    libguestfs-tools
+    libguestfs-tools \
+    --force-ignore-with-dependencies '^(kernel-|linux-firmware)' \
+    --force-ignore-with-dependencies '^(python[3]{0,1}-|perl[3]{0,1}-)' \
+    --force-ignore-with-dependencies '^(mesa-|libwayland-|selinux-policy|mozjs60)' \
```
Comment (andreabolognani) [inline hack/rpm-deps.sh:108]: Isn't dropping SELinux-related packages a bit too aggressive? Are you sure that's not going to cause any issues?

Comment (rmohr) [reply hack/rpm-deps.sh:108]: I am looking forward to the e2e test results with libguestfs :)

**Comment by rmohr** [comment]
/retest

@alicefr test results look good, can you have another look?

File: hack/rpm-deps.sh (hunk 1 above)
Comment (alicefr) [reply hack/rpm-deps.sh:108]: @andreabolognani I think it should be fine to drop selinux packages. Selinux runs outside the container not inside.

**Comment by alicefr** [comment]
it looks good. I also tried a couple of libguestfs-tools it seems working fine :)
/lgtm

**Comment by alicefr** [comment]
The diet was also very efficient 1.3G -> 605 MB ;)

**Comment by alicefr** [comment]
@rmohr the storage tests are failing and they are the guestfs tests. However, it is because the pod is still pending. So, I think it is because the pvc hasn't be bound and it times out. I'll trigger it once more
/retest

**Comment by alicefr** [comment]
ok the storage tests are still failing but now the failed tests are not the guestfs tests. So, it was just flaky

**Comment by rmohr** [comment]
> ok the storage tests are still failing but now the failed tests are not the guestfs tests. So, it was just flaky

Yes. Something is wrong with them since a few days.

**Comment by rmohr** [comment]
@alicefr rebased. Could you restore the lgtm?

**Pull request merged**